modules/gapi/include/opencv2/gapi/gtransform.hpp

```diff
@@ -0,0 +1,210 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2019 Intel Corporation
+
+#ifndef OPENCV_GAPI_GTRANSFORM_HPP
+#define OPENCV_GAPI_GTRANSFORM_HPP
+
+#include <functional>
+#include <type_traits>
+#include <utility>
+
+#include <opencv2/gapi/gcommon.hpp>
+#include <opencv2/gapi/util/util.hpp>
+#include <opencv2/gapi/garg.hpp>
+#include <opencv2/gapi/gtype_traits.hpp>
+#include <opencv2/gapi/util/compiler_hints.hpp>
+
+namespace cv
+{
+
+template <typename T>
+struct TransType;
+template <>
+struct TransType<cv::GMat>
+{
+    using type = cv::GMat;
+};
+template <>
+struct TransType<cv::GMatP>
+{
+    using type = cv::GMat;
+};
+template <>
+struct TransType<cv::GScalar>
+{
+    using type = cv::GScalar;
+};
+template <typename U>
+struct TransType<cv::GArray<U>>
+{
+    using type = cv::GArray<U>;
+};
+template <typename T>
+struct TransType
+{
+    using type = T;
+};
+
+template <typename T>
+using is_nongapi_type2 = std::is_same<T, typename TransType<T>::type>;
```

Comment:
Why `is_nongapi_type ` from gkernel.hpp cannot be utulized here?

Reply:
Approach has been reconsidered

Reply:
Done! Please, resolve the conversation.